File: Packages/KodCore/Tests/LanguageClientTests/Fixtures/SourceKitLSPFixture/Sources/SourceKitLSPFixture/Greeter.swift
/// A tiny, deterministic fixture type for SourceKit-LSP integration tests:
/// hover, definition, references, document symbols, diagnostics, and
/// semantic tokens are all exercised against this single small file.
public struct Greeter {
    public let name: String

    public init(name: String) {
        self.name = name
    }

    /// Returns a friendly greeting for `name`.
    public func greet() -> String {
        "Hello, \(name)!"
    }
}

public func makeDefaultGreeter() -> Greeter {
    Greeter(name: "Kod")
}

File: Packages/KodCore/Tests/LanguageAdaptersTests/Fixtures/PythonFixture/greeter.py
"""A tiny, deterministic fixture for Pyright integration tests: hover,
definition, references, and document symbols are all exercised against
this single small file."""


class Greeter:
    def __init__(self, name: str) -> None:
        self.name = name

    def greet(self) -> str:
        """Returns a friendly greeting for this Greeter's name."""
        return f"Hello, {self.name}!"


def make_default_greeter() -> Greeter:
    return Greeter("Kod")
